Jetsnack/app/src/main/java/com/example/jetsnack/ui/home/Cart.kt

```diff
@@ -17,19 +17,329 @@
+) {
+    val (count, updateCount) = remember { mutableStateOf(1) }
+    ConstraintLayout(
         modifier = modifier
-            .fillMaxSize()
-            .wrapContentSize()
-    )
+            .fillMaxWidth()
+            .clickable { onSnackClick(snack.id) }
+            .padding(horizontal = 24.dp)
+    ) {
+        val (divider, image, name, tag, priceSpacer, price, remove, quantity) = createRefs()
+        createVerticalChain(name, tag, priceSpacer, price, chainStyle = ChainStyle.Packed)
+        SnackImage(
+            imageUrl = snack.imageUrl,
+            modifier = Modifier
+                .preferredSize(100.dp)
+                .constrainAs(image) {
+                    top.linkTo(parent.top, margin = 16.dp)
+                    bottom.linkTo(parent.bottom, margin = 16.dp)
+                    start.linkTo(parent.start)
+                }
+        )
+        Text(
+            text = snack.name,
+            style = MaterialTheme.typography.subtitle1,
+            color = JetsnackTheme.colors.textSecondary,
+            modifier = Modifier.constrainAs(name) {
+                linkTo(
+                    start = image.end,
+                    startMargin = 16.dp,
+                    end = remove.start,
+                    endMargin = 16.dp,
+                    bias = 0f
+                )
+            }
+        )
+        IconButton(
+            onClick = { /* todo */ },
+            modifier = Modifier.constrainAs(remove) {
+                top.linkTo(parent.top)
+                end.linkTo(parent.end)
+            }.padding(top = 12.dp)
+        ) {
+            Icon(asset = Icons.Filled.Close, tint = JetsnackTheme.colors.brand)
+        }
+        Text(
+            text = snack.tagline,
+            style = MaterialTheme.typography.body1,
+            color = JetsnackTheme.colors.textHelp,
+            modifier = Modifier.constrainAs(tag) {
+                linkTo(
+                    start = image.end,
+                    startMargin = 16.dp,
+                    end = parent.end,
+                    endMargin = 16.dp,
+                    bias = 0f
+                )
+            }
+        )
+        Spacer(
+            Modifier
+                .preferredHeight(8.dp)
+                .constrainAs(priceSpacer) {
+                    linkTo(top = tag.bottom, bottom = price.top)
+                }
+        )
+        Text(
+            text = "$12.99",
+            style = MaterialTheme.typography.subtitle1,
+            color = JetsnackTheme.colors.textPrimary,
+            modifier = Modifier.constrainAs(price) {
+                linkTo(
+                    start = image.end,
+                    end = quantity.start,
+                    startMargin = 16.dp,
+                    endMargin = 16.dp,
+                    bias = 0f
+                )
+            }
+        )
+        QuantitySelector(
+            count = count,
+            decreaseItemCount = { if (count > 0) updateCount(count - 1) },
```

Comment:
Maybe add a comment that this logic should really live in the ViewModel/data layer